src/templates/dotnet/executeDotnetTemplateCommand.ts

```diff
@@ -36,3 +38,36 @@ export function getDotnetItemTemplatePath(runtime: ProjectRuntime): string {
 export function getDotnetProjectTemplatePath(runtime: ProjectRuntime): string {
     return path.join(getDotnetTemplatesPath(), `projectTemplates-${runtime}.nupkg`);
 }
+
+let cachedFramework: string | undefined;
+async function getFramework(workingDirectory: string | undefined): Promise<string> {
+    if (!cachedFramework) {
+        let versions: string = '';
+        try {
+            versions += await cpUtils.executeCommand(undefined, workingDirectory, 'dotnet', '--version');
+        } catch {
+            // ignore
+        }
+
+        try {
+            versions += await cpUtils.executeCommand(undefined, workingDirectory, 'dotnet', '--list-sdks');
+        } catch {
+            // ignore
+        }
+
+        const majorVersions: string[] = ['3', '2'];
+        for (const majorVersion of majorVersions) {
+            const regExp: RegExp = new RegExp(`^\\s*${majorVersion}\\.`);
+            if (regExp.test(versions)) {
+                cachedFramework = `netcoreapp${majorVersion}.0`;
```

Comment:
What's the point of getting the sdk version if you just end up setting cachedFramework to either 3 or 2?

Reply:
I'm getting the sdk version to see if they have 3 or 2 on their machine. If they don't, it'll throw an error below and never set the cachedFramework

Reply:
`--version` doesn't get the major version? 😮

Reply:
Users can have multiple versions of the sdk installed on their machine and `dotnet --version` will only print out one version. Usually that's the latest sdk version, but users can configure it to be any version.

Here's example output:
```
> dotnet --version
2.1.202
> dotnet --list-sdks
1.1.14 [/usr/local/share/dotnet/sdk]
2.1.202 [/usr/local/share/dotnet/sdk]
```